### PR TITLE
B-19763 I-12974 mileage does not updated to correspond with zip code when destination address is updated

### PR DESCRIPTION
**Pull request by WeatherfordAaron**
## TL;DR:
When HHG shipment has no approved Destination SIT service items:
- You create a Payment Request for Linehaul/Shorthaul and FSC then Update the Destination Address for the shipment and approve the Destination Address Update, then create a new Payment Request.
     **The destination ZIP for the Payment Request as well as the Calculations should change to reflect the change**
- You add and approve Destination SIT service items and then Update the Destination Address for the shipment and approve the Destination Address Update, then create a new Payment Request for Linehaul/Shorthaul, FSC, Destination SIT Delivery, and Destination SIT FSC.
     **The destination ZIP and associated calculations for the Linehaul/Shorthaul and FSC should terminate at the destination ZIP as it was at the time that the Destination SIT service items were approved.**
     **The destination ZIP and associated calculations for the Destination SIT Delivery and Destination SIT FSC should terminate at the destination of the shipment.**

## What was causing the issue?
1. Create HHG move
2. SC submits move
3. TOO approves move
4. Prime updates shipment - add weight values and save
5. Prime creates payment request (add Domestic linehaul & FSC)
6. TIO reviews payment request and authorizes payment
7. TOO can view the Service Items/Calculations 
8. Prime update Destination address to an address greater than 50 miles
9. TOO approves address update
10. Prime creates payment request (add Domestic linehaul & FSC)
11. TIO reviews payment request

Issue: The updated address zip is used, but the mileage calculation is not updated and remained the same as viewed in step 6/7 above

This PR should fix that so that the calculations update in this instance. It's also important that the functionality from the original PR is not messed up. You can test that by performing the following steps:

1. As a Customer, create a new move with HHG shipment and submit to Service Counsellor.
2. As Service Counsellor, update the orders and Approve/Send the move on to the next steps.
3. As TOO/TIO, approve the HHG shipment.
4. As Prime, add the Destination SIT service item.
5. As TOO/TIO,approve the Destination SIT service item bundle (should be a few Destination SIT service items) that was created.
6. As Prime, Update Shipment to include dates and weights.
7. As Prime, create Payment Requests for:
- Linehaul (or Shorthaul)
- Fuel Surcharge
- Destination SIT Delivery
- Destination SIT Fuel Surcharge
8. As TOO/TIO, navigate to "Payment Requests" tab and click the "Review Payment Requests" button.
9. You don't necessarily have to Approve or Reject any of the Payment Requests., but you should note the From and To Zip Codes under the Mileage section for each set of Service Item Calculations. The shipment-level Service Items should show the same "To" Zip as the shipment's destination address. The Destination SIT Service Items should have the Zip for the SIT location in the "From" Zip value and the Zip for the final delivery address in the "To" Zip value.
10. As Prime, update the Shipment Destination Address. You can change what you want but the main item to change is the Zip value. Please change this to a Zip that is a difference of at least 50 miles from the Zip you are changing it from.
11. As TOO/TIO, you should see a notification for needing to approve the destination address update. Go ahead and approve this update. If you did not Approve or Reject the previous Payment Requests, you should notice that none of the Addresses for those Payment Requests have changed.
12. As Prime, create new Payment Requests for:
- Linehaul (or Shorthaul)
- Fuel Surcharge
- Destination SIT Delivery
- Destination SIT Fuel Surcharge
13. As TOO/TIO, navigate to "Payment Requests" tab and click the "Review Payment Requests" button.
14. You don't necessarily have to Approve or Reject any of the Payment Requests. However, you will need to review the correct payment requests (the ones that were just submitted).  You should note the From and To Zip Codes under the Mileage section for each set of Service Item Calculations. The shipment-level Service Items should show the Zip of the Destination SIT from when the Destination SIT Service Items were approved. The Destination SIT Service Items should have the Zip for the SIT location in the "From" Zip value and the Zip for the final delivery address of the shipment (the new address that was approved) in the "To" Zip value.

### CODE REVIEW

File: migrations/app/migrations_manifest.txt [hunk 1]
```diff
@@ -951,6 +951,6 @@
 20240531153321_update_tio_role_name.up.sql
 20240531154303_add_more_submitted_columns_to_ppm_document_tables.up.sql
 20240603040207_add_submitted_cols_to_moving_expenses.up.sql
-20240604203456_add_effective_expiraton_date_ghc_fuel.up.sql
 20240603152949_update_too_role_name.up.sql
+20240604203456_add_effective_expiraton_date_ghc_fuel.up.sql
```
Comment (WeatherfordAaron) [inline migrations/app/migrations_manifest.txt:955]: automatically changed by hook

File: pkg/payment_request/service_param_value_lookups/distance_zip_lookup.go [hunk 2]
```diff
@@ -42,7 +42,19 @@ func (r DistanceZipLookup) lookup(appCtx appcontext.AppContext, keyData *Service
 
 	// Now calculate the distance between zips
 	pickupZip := r.PickupAddress.PostalCode
-	destinationZip := r.DestinationAddress.PostalCode
+	var destinationZip string
```
Comment (WeatherfordAaron) [inline pkg/payment_request/service_param_value_lookups/distance_zip_lookup.go:45]: this is where the logic forks off from normal operation and works out conditions that would possibly be related to this ticket

File: pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go [hunk 3]
```diff
@@ -522,40 +566,12 @@ func getDestinationAddressForService(appCtx appcontext.AppContext, serviceCode m
 	case models.ReServiceCodeDPK, models.ReServiceCodeDNPK:
 		// Destination address isn't needed
 		return models.Address{}, nil
-	case models.ReServiceCodeDLH, models.ReServiceCodeDSH, models.ReServiceCodeFSC:
```
Comment (WeatherfordAaron) [inline pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go:525]: This is removed code from the original work done on the associated story. This logic was reworked into another method

File: pkg/services/ppmshipment/aoa_packet_creator_test.go [hunk 4]
```diff
@@ -269,6 +269,11 @@ func (suite *PPMShipmentSuite) TestSaveAOAPacket() {
 				appCtx := suite.AppContextForTest()
 
 				ppmShipment := factory.BuildPPMShipment(nil, []factory.Customization{
+					{
```
Comment (WeatherfordAaron) [inline pkg/services/ppmshipment/aoa_packet_creator_test.go:272]: Added this to see if it would help my tests pass. I don't think it necessarily has to be there but it isn't hurting anything.

File: pkg/testdatagen/scenario/subscenarios.go [hunk 5]
```diff
@@ -687,7 +687,6 @@ func subScenarioPaymentRequestCalculations(
 		createTXO(appCtx)
 		createTXOUSMC(appCtx)
 
-		// For displaying the Domestic Line Haul calculations displayed on the Payment Requests and Service Item review page
```
Comment (WeatherfordAaron) [inline pkg/testdatagen/scenario/subscenarios.go:690]: removed inaccurate comment

File: pkg/testdatagen/scenario/subscenarios.go [hunk 6]
```diff
@@ -696,11 +695,9 @@ func subScenarioPaymentRequestCalculations(
 				Status: models.MTOShipmentStatusApproved,
 			},
 		)
-		// Locator PARAMS
```
Comment (WeatherfordAaron) [inline pkg/testdatagen/scenario/subscenarios.go:699]: removed confusing comments

File: playwright/tests/office/primesimulator/primeSimulatorFlows.spec.js [hunk 7]
```diff
@@ -102,9 +102,67 @@ test.describe('Prime simulator user', () => {
     await expect(page.getByText('Estimated Weight:7500')).toBeVisible();
     await expect(page.getByText('Actual Weight:8000')).toBeVisible();
     await expect(page.getByText('Destination Address:142 E Barrel Hoop Circle, Joshua Tree, CA 92252')).toBeVisible();
+  });
 
-    // Can only create a payment request if there is a destination
+  test('is able to create payment requests for shipment-level service items', async ({ page, officePage }) => {
```
Comment (WeatherfordAaron) [inline playwright/tests/office/primesimulator/primeSimulatorFlows.spec.js:107]: modified the test that was here (since it was no longer needed) to this test (which is now needed to protect the story)

File: src/components/Customer/PPM/Booking/DateAndLocationForm/DateAndLocationForm.jsx [hunk 8]
```diff
@@ -86,13 +88,38 @@ const DateAndLocationForm = ({ mtoShipment, destinationDutyLocation, serviceMemb
...
+    }
+    if (values.closeoutOffice) {
+      meta = '';
+    }
+    return {};
+  };
+
   return (
-    <Formik initialValues={initialValues} validationSchema={Yup.object().shape(validationShape)} onSubmit={onSubmit}>
+    <Formik
```
Comment (WeatherfordAaron) [inline src/components/Customer/PPM/Booking/DateAndLocationForm/DateAndLocationForm.jsx:114]: Big breakthroughs on this is to add the validateOnBlur, validateOnMount, and validateOnChange attributes to the Formik element.

File: src/components/Customer/PPM/Booking/DateAndLocationForm/DateAndLocationForm.test.jsx [hunk 9]
```diff
@@ -1,7 +1,6 @@
 import React from 'react';
```
Comment (WeatherfordAaron) [inline src/components/Customer/PPM/Booking/DateAndLocationForm/DateAndLocationForm.test.jsx:1]: For this file, added a lot of act wrappers and WaitFor wrappers to assist in keeping this test file from becoming flaky at some point

File: src/components/LocationSearchBox/LocationSearchBox.jsx [hunk 10]
```diff
@@ -175,6 +175,10 @@ export const LocationSearchBoxComponent = ({
     }
   };
 
+  const handleFocus = () => {
```
Comment (WeatherfordAaron) [inline src/components/LocationSearchBox/LocationSearchBox.jsx:178]: This addition allows us to trigger validation more readily on the closeout field

File: src/components/Office/RequestedShipments/SubmittedRequestedShipments.jsx [hunk 11]
```diff
@@ -284,7 +284,7 @@ SubmittedRequestedShipments.propTypes = {
   allowancesInfo: PropTypes.shape({
     branch: PropTypes.string,
     grade: PropTypes.string,
-    totalWeight: PropTypes.string,
+    totalWeight: PropTypes.number,
```
Comment (WeatherfordAaron) [inline src/components/Office/RequestedShipments/SubmittedRequestedShipments.jsx:287]: There was a warning for a type mismatch here so I corrected that

File: src/components/Office/ShipmentCustomerSIT/ShipmentCustomerSIT.test.jsx [hunk 12]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, act } from '@testing-library/react';
```
Comment (WeatherfordAaron) [inline src/components/Office/ShipmentCustomerSIT/ShipmentCustomerSIT.test.jsx:2]: added an act wrapper in this file due to a warning that was present

File: src/components/Office/ShipmentForm/ShipmentForm.test.jsx [hunk 13]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable react/jsx-props-no-spreading */
```
Comment (WeatherfordAaron) [inline src/components/Office/ShipmentForm/ShipmentForm.test.jsx:1]: Added act wrappers and waitFor wrappers due to warnings received in CircleCI. Ultimately, this may help to keep this test from becoming flaky.

**Review by brooklyn-welsh — APPROVED** [review]
Was able to follow steps and test that it works as stated. LGTM ✅

**Review by deandreJones — APPROVED** [review]

**Review by JamesHawks224 — APPROVED** [review]
testing: success
code: lgtm
